core/src/main/java/com/linecorp/armeria/common/metric/MoreMeters.java

```diff
@@ -160,16 +159,32 @@ public static Timer newTimer(MeterRegistry registry, String name, Iterable<Tag>
         final Duration minExpectedValue =
                 minExpectedValueNanos != null ? Duration.ofNanos(minExpectedValueNanos.longValue()) : null;
 
-        return Timer.builder(name)
-                    .tags(tags)
-                    .maximumExpectedValue(maxExpectedValue)
-                    .minimumExpectedValue(minExpectedValue)
-                    .publishPercentiles(distStatsConfig.getPercentiles())
-                    .publishPercentileHistogram(distStatsConfig.isPercentileHistogram())
-                    .percentilePrecision(distStatsConfig.getPercentilePrecision())
-                    .distributionStatisticBufferLength(distStatsConfig.getBufferLength())
-                    .distributionStatisticExpiry(distStatsConfig.getExpiry())
-                    .register(registry);
+        final Timer.Builder builder =
```

Comment:
The new logic looks good 👍 

Is it possible to also update `MoreMeters#newDistributionSummary` to have similar flow? i.e. Update `[minimum|maximum]ExpectedValue` outside of the if branch, early return if `slas` is null, etc..

Reply:
I tried to refactor the code, but the fact that we can't use the same type for `maximumExpectedValue`, `minimumExpectedValue` and `slo|serviceLevelObjectiveBoundaries` across all micrometer version makes it hard to refactor the code in the same way.

The best I can get is like the following...:
```java
        if (MICROMETER_1_5) {
            return builder.maximumExpectedValue(distStatsConfig.getMaximumExpectedValueAsDouble())
                          .minimumExpectedValue(distStatsConfig.getMinimumExpectedValueAsDouble())
                          .serviceLevelObjectives(distStatsConfig.getServiceLevelObjectiveBoundaries())
                          .register(registry);
        }
        final Double maxExpectedValueNanos = distStatsConfig.getMaximumExpectedValueAsDouble();
        final Double minExpectedValueNanos = distStatsConfig.getMinimumExpectedValueAsDouble();
        final Long maxExpectedValue =
                maxExpectedValueNanos != null ? maxExpectedValueNanos.longValue() : null;
        final Long minExpectedValue =
                minExpectedValueNanos != null ? minExpectedValueNanos.longValue() : null;
        builder.maximumExpectedValue(maxExpectedValue);
        builder.minimumExpectedValue(minExpectedValue);
        final double[] slas = distStatsConfig.getServiceLevelObjectiveBoundaries();
        if (slas != null) {
            builder.sla(Arrays.stream(slas).mapToLong(sla -> (long) sla).toArray());
        }
        return builder.register(registry);
```
...but it looks no better than the original one.

Reply:
> We can't use the same type for maximumExpectedValue, minimumExpectedValue

I see, I thought due to the following fix, we can use `Double` across major versions
https://github.com/micrometer-metrics/micrometer/pull/2165

If this is an issue, I guess it's fine to keep as-is. I'll check one more time before merging.

Reply:
The latest version of Micrometer is 1.12.5. I think we can drop support for `< 1.5` versions to reduce maintenance costs.

Reply:
Yeah, I think we can drop that.

Reply:
I will modify the code to drop `< 1.5` support.